src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -801,6 +809,15 @@ void GcsActorManager::DestroyActor(const ActorID &actor_id,
         // Destroy placement group owned by this actor.
         destroy_owned_placement_group_if_needed_(actor_id);
       }));
+
+  // Inform all creation callbacks that the actor was cancelled, not created.
+  auto iter = actor_to_create_callbacks_.find(actor_id);
+  if (iter != actor_to_create_callbacks_.end()) {
+    for (auto &callback : iter->second) {
+      callback(actor, rpc::PushTaskReply());
+    }
+    actor_to_create_callbacks_.erase(iter);
```

Comment:
Ah good point about leaking the callbacks! 🤦 

>Maybe keep the raise location as it is, but we can store a list of callbacks up there and use it here?

I was trying to limit the scope of this change to cancellation, but should the callbacks also be invoked for any of the early-return cases? This was my thinking for each case:
1. Actor doesn't exist - actor creation callback shouldn't exist since it was never registered.
2. Actor is already dead - this should only happen if actor creation has already succeeded.

Let me know if I'm missing anything there. Otherwise, I'll do as you suggest, erase the callbacks from the `actor_to_create_callbacks_` map early and invoke the callbacks at the end.